src/test/java/core/basesyntax/HelloWorldTest.java

```diff
@@ -1,7 +1,189 @@
+    }
+
+    @Test
+    public void reader_correctFile_Ok() {
+        List<String> expected = new ArrayList<>();
+        expected.add("b,banana,45");
+        List<String> actual = reader.read("src/test/java/core/basesyntax/recources/inputFile.csv");
+        Assert.assertEquals("Readed data is wrong.", expected.toString(), actual.toString());
+    }
+
+    @Test
+    public void reader_incorrectFile_NotOk() {
+        Assert.assertThrows(RuntimeException.class, () -> {
+            reader.read("src/test/java/core/basesyntax/recources/incorrectFile.csv");
+        });
+    }
+
+    @Test
+    public void reader_emptyFile_NotOk() {
+        Assert.assertThrows(RuntimeException.class, () -> {
+            reader.read("src/test/java/core/basesyntax/recources/emptyFile.csv");
+        });
+    }
+
+    @Test
+    public void reader_incorrectData_NotOk() {
+        Assert.assertThrows(RuntimeException.class, () -> {
+            reader.read("src/test/java/core/basesyntax/recources/incorrectData.csv");
+        });
+    }
+
+    @Test
+    public void converter_correctData_Ok() {
+        List<String> data = reader.read("src/test/java/core/basesyntax/recources/inputFile.csv");
+        List<FruitTransaction> actual = dataConverter.convertToTransaction(data);
+        List<FruitTransaction> expected = soloListFruit;
+        Assert.assertEquals(1, actual.size());
+        Assert.assertEquals(actual.get(0), expected.get(0));
+    }
+
+    @Test
+    public void converter_incorrectData_NotOk() {
+        List<String> data = reader.read(
+                "src/test/java/core/basesyntax/recources/incorrectDataForConvert.csv");
+        Assert.assertThrows(RuntimeException.class, () -> {
+            dataConverter.convertToTransaction(data);
+        });
+    }
+
+    @Test
+    void shopService_oneDatacorrect_Ok() {
+        Map<String, Integer> expected = new HashMap<>();
+        expected.put("banana", 45);
+        Map<String, Integer> actual = shopService.process(soloListFruit);
+        Assert.assertEquals(1, actual.size());
+        Assert.assertEquals(actual, expected);
+    }
+
+    @Test
+    void shopService_multiplyDataCorrect_Ok() {
+        multiplyListFruits.clear();
+        multiplyListFruits.add(new FruitTransaction("b", "banana", 45));
+        multiplyListFruits.add(new FruitTransaction("p", "banana", 15));
+        multiplyListFruits.add(new FruitTransaction("s", "banana", 25));
+        multiplyListFruits.add(new FruitTransaction("r", "banana", 15));
+        Map<String, Integer> expected = new HashMap<>();
+        expected.put("banana", 70);
+        Map<String, Integer> actual = shopService.process(multiplyListFruits);
+        Assert.assertEquals(1, actual.size());
+        Assert.assertEquals(actual, expected);
+    }
+
+    @Test
+    void shopService_NegativeBalance_NotOk() {
+        soloListFruit.clear();
+        soloListFruit.add(new FruitTransaction("b", "banana", -15));
```

Comment:
either perform  soloListFruit.clear(); in @afterEach annotated method or use method level variable, there in no need to make it constant